lib/util/esWrapper.js

```diff
@@ -154,21 +154,76 @@ class ESWrapper {
       }
     }
 
-    switch (error.meta.statusCode) {
-      case 400:
-        kuzzleError = kuzzleError || this._handleBadRequestError(error, message);
+    switch (error.displayName) {
+      case 'BadRequest':
+        if (kuzzleError) {
+          break;
+        }
+
+        if (error.body && error.body.error) {
+          message = error.body.error.root_cause
+            ? error.body.error.root_cause[0].reason
+            : error.body.error.reason;
+        }
+
+        kuzzleError = errorsManager.getError(
+          'external',
+          'elasticsearch',
+          'unexpected_bad_request_error',
+          message);
+
+        debug('unhandled "BadRequest" elasticsearch error: %a', error);
+
         break;
 
-      case 404:
-        kuzzleError = kuzzleError || this._handleNotFoundError(error, message);
+      case 'NotFound':
+        if (kuzzleError) {
+          break;
+        }
+
+        if (error.body && error.body.error) {
+          message = error.body.error
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Correct this attempt to modify "message" or use "let" in its declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3500)